modules/client/shared/src/main/scala/com/azavea/stac4s/api/client/PaginationToken.scala

```diff
@@ -0,0 +1,15 @@
+package com.azavea.stac4s.api.client
+
+import eu.timepit.refined.types.numeric.PosInt
+import io.circe.generic.semiauto._
+import io.circe.refined._
+import io.circe.{Decoder, Encoder}
+
+import java.time.Instant
+
+final case class PaginationToken(timestampAtLeast: Instant, serialIdGreaterThan: PosInt)
```

Comment:
This is correct for Franklin's implementation of pagination, but only for Franklin's -- the particular pagination strategy is left open to implementers. So there are a few things that are a bit weird as a result --

- client side search filters can't assume that the value in `next` will be a json `PaginationToken`, since it will just be an opaque string by the time it reaches the client. This means this is probably wrong for any other STAC API implementation.
- the thing Franklin actually sends back in this field is a base64 encoding of the pagination token, so decoding it probably won't work client side (not sure if you had a chance to test that)
- the thing Franklin expects is for the client to have a hold of the opaque token, so it does a base64 decode to get the json out

https://github.com/azavea/franklin/blob/f5be8ddf48661c5bc43cbd22cb7277e961641803/application/src/main/scala/com/azavea/franklin/api/schemas/package.scala#L84-L85

So I think this shouldn't work

Reply:
I'd say this is _fine_; since it is a default implementation (i.e. if smth doesn't work user can always define theirs own STACClient with a redefined search filters: see https://github.com/azavea/stac4s/blob/7f6abceeebea59dfc9a605891c73fc121116610b/modules/client/jvm/src/main/scala/com/azavea/stac4s/api/client/SttpStacClient.scala#L9-L13 - they only need to define their own version of `SearchFilters`), if smth won't work in the future with other STAC Services I think we should handle it later (?).

Reply:
Opened an issue for that #198

Reply:
I'm not really talking about streaming here though -- I don't think this pagination encoding will work, and the tests included here don't exercise pagination. Since the pagination implementation here isn't exercised and I think won't work, I think it should just be omitted until someone is working on #198

Reply:
Oops forgot to push _not tested_ codecs

Reply:
@jisantuc added it into the issue description https://github.com/azavea/stac4s/issues/198